### PR TITLE
enhancement: allow relative links in the Blocks editor

### PR DESCRIPTION
**Pull request by Ben-Zahler**
### What does it do?

Allows authors to create relative links in the blocks editor

### Why is it needed?

Currently, links cannot be saved when entering relative links in the blocks editor.

### How to test it?

- open the new blocks editor
- create a link
- enter any relative link as the link target (e.g. /about-strapi)
expected: the user can save the link and save the block

### Related issue(s)/PR(s)

https://github.com/strapi/strapi/issues/18562

### CODE REVIEW

**Comment by Ben-Zahler** [comment]
@remidej this is a follow up to the PR I raised last week. For me all other cases in https://github.com/strapi/strapi/issues/18562 were working when I tested this on the 4.17 branch.

**Review by remidej — CHANGES_REQUESTED** [review]
Hey, thanks for following up. I wasn't sure initially, but I think it's a good idea to support relative links indeed.

There's just 1 thing to add. The validations also happen in the backend, so I believe here even though the admin will let you add the link, you won't be able to save the content.

Could you also update the validations in the backend? The idea would be to use a custom yup.string().test to do the same as the frontend, instead of yup.string().url(). It happens [in the blocks-validators file](https://github.com/strapi/strapi/blob/334c95408abdd9ecc0ece0e8c0fd5eb0024ad027/packages/core/strapi/src/services/entity-validator/blocks-validator.ts#L23).

**Comment by Ben-Zahler** [comment]
@remidej when I tested this on the 4.17 branch, backend validations did not kick in and I was able to save.
As mentioned, I did not investigate on the validation but was happy that it worked :-)

File: packages/core/strapi/src/services/entity-validator/blocks-validator.ts [hunk 1]
```diff
@@ -18,14 +18,27 @@ const textNodeValidator = yup.object().shape({
...
-  url: yup.string().url().required(),
+  url: yup
+    .string()
+    .test('invalid-url', 'Please specify a valid link.', (value) => checkValidLink(value ?? '')),
   children: yup.array().of(textNodeValidator).required(),
 });
 
 // TODO: remove any with a correct Type
 const inlineNodeValidator: any = yup.lazy((value: { type: string }) => {
+  console.log(`validating online node of of ${value.type}`);
```
Comment (remidej) [inline packages/core/strapi/src/services/entity-validator/blocks-validator.ts:41]: can you remove this log?

**Comment by Ben-Zahler** [comment]
@remidej @innerdvations the backend validation should now be implemented as suggested by @remidej .
I hope the linter passes, I don't trust my setup 100%. Do the test have to be triggered manually?

**Comment by lzhou-aops** [comment]
I would love to see this added as well!

**Comment by Ben-Zahler** [comment]
hi @remidej , thanks for the Preview deployment, is there anything esle I can do to support this PR?

**Review by remidej — APPROVED** [review]

**Pull request merged**